src/Network/NWTxtRecord.cs

```diff
@@ -0,0 +1,240 @@
+//
+// NWTxtRecord.cs: Bindings the Netowrk nw_txt_record_t API
+//
+// Authors:
+//   Miguel de Icaza (miguel@microsoft.com)
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Text;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using nw_advertise_descriptor_t=System.IntPtr;
+using OS_nw_advertise_descriptor=System.IntPtr;
+using OS_nw_txt_record=System.IntPtr;
+
+
+namespace Network {
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public enum NWTxtRecordFindKey {
+		Invalid = 0,
+		NotPresent = 1,
+		NoValue = 2,
+		EmptyValue = 3,
+		NonEmptyValue = 4
+	}
+	
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public class NWTxtRecord : NativeObject {
+		public NWTxtRecord (IntPtr handle, bool owns) : base (handle, owns) { }
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr nw_txt_record_create_with_bytes (IntPtr txtBytes, nuint len);
+
+		public static NWTxtRecord FromBytes (byte [] bytes)
+		{
+			if (bytes == null || bytes.Length == 0)
+				return null;
+			unsafe {
+				fixed (byte *p = &bytes[0]){
+					var x = nw_txt_record_create_with_bytes ((IntPtr) p, (nuint) bytes.Length);
+					if (x == IntPtr.Zero)
+						return null;
+					return new NWTxtRecord (x, owns: true);
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr nw_txt_record_create_dictionary();
+
+		public static NWTxtRecord CreateDictionary ()
+		{
+			var x = nw_txt_record_create_dictionary ();
+			if (x == IntPtr.Zero)
+				return null;
+			return new NWTxtRecord (x, owns: true);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr nw_txt_record_copy (IntPtr other);
+		
+		public NWTxtRecord Clone ()
+		{
+			if (Handle == IntPtr.Zero)
+				return null;
+			return new NWTxtRecord (nw_txt_record_copy (Handle), owns: true);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern NWTxtRecordFindKey nw_txt_record_find_key (IntPtr handle, string key);
+		
+		public NWTxtRecordFindKey FindKey (string key)
+		{
+			return nw_txt_record_find_key (Handle, key);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern byte nw_txt_record_set_key(IntPtr handle, string key, IntPtr value, nuint valueLen);
+
+		public bool SetKey (string key, byte [] value)
+		{
+			if (value == null)
+				return nw_txt_record_set_key (Handle, key, IntPtr.Zero, 0) != 0;
+			else {
```

Comment:
Redundant else statement that causes unnecessary indentation. You don't need an else because the if statement returns.